utils/python/CIME/XML/GenericXML.py

```diff
@@ -119,6 +127,6 @@ def get_resolved_value(self, raw_value):
             ref = self.get_value(var)
             if(ref is not None):
                 logging.debug("resolve: "+ref)
-                item_data = item_data.replace(m.group(), ref)
+                item_data = item_data.replace(m.group(), self.get_resolved_value(ref))
```

Comment:
This was the only significant behavioral change. I was getting unresolved values because ref itself had refs to other things.